src/main/java/com/thealgorithms/stacks/CelebrityFinder.java

```diff
@@ -0,0 +1,40 @@
+package com.thealgorithms.stacks;
+
+/**
+ * Solves the celebrity problem using a stack-based algorithm.
+ *
+ * <p>Celebrity is someone known by everyone but doesn't know anyone else.
+ * <p>Applications: Graph theory and social network analysis.
+ *
+ * @author Hardvan
+ */
+public final class CelebrityFinder {
+    private CelebrityFinder() {
+    }
+
+    /**
+     * Finds the celebrity in the given party matrix.
+     *
+     * @param party A 2D matrix where party[i][j] is 1 if i knows j, otherwise 0.
+     * @return The index of the celebrity, or -1 if there is no celebrity.
+     */
+    public static int findCelebrity(int[][] party) {
+        int n = party.length;
```

Comment:
Will be better to using directly party.length

Reply:
Done!